### PR TITLE
maxCountTires

### PR DESCRIPTION
**Pull request by RomchikSt**
Trello:https://trello.com/c/3sC8Q3tr/73-item-details-max-count

After:

https://github.com/unreal998/top-kolesa-fe/assets/140477189/cc11391d-15ed-4914-ba9d-19393a174178

### CODE REVIEW

**Comment by unreal998** [comment]
Add some small tooltip with the message like 'you have rich a maximum count of available tires in our warehouse'

**Comment by RomchikSt** [comment]
Added:

https://github.com/unreal998/top-kolesa-fe/assets/140477189/ccb5ac99-42d9-464e-99cf-f543337926eb

**Comment by unreal998** [comment]
Perfect)

File: src/modules/itemDetailsPage/components/BuyOptions.tsx [hunk 1]
```diff
@@ -88,6 +105,17 @@ export default function BuyOptions({ tireId }: { tireId: number | undefined }) {
...
 
+  const handleTooltipClose = () => {
+    setTooltipOpen(false);
+  };
+
+  const handleTooltipOpen = () => {
+    setTooltipOpen(numberOfTires === selectedItemData?.in_stock);
+    setTimeout(() => {
+      setTooltipOpen(false);
+    }, 4000);
```
Comment (unreal998) [inline src/modules/itemDetailsPage/components/BuyOptions.tsx]: move animation timeout to the separate constant

File: src/shared/components/header/CartItem.tsx [hunk 2]
```diff
@@ -54,15 +58,26 @@ export const CartItem: React.FC<CartItemProps> = ({
...
-          ? { ...item, numberOfTires: item.numberOfTires + 1 }
-          : item,
-      );
-      localStorage.setItem('cartItem', JSON.stringify(updatedCartItems));
-      updateCartItems(updatedCartItems);
+      if (cartItemData.numberOfTires === selectedItemData?.in_stock) {
+        setTooltipOpen(true);
+        setTimeout(() => {
+          setTooltipOpen(false);
+        }, 4000);
```
Comment (unreal998) [inline src/shared/components/header/CartItem.tsx]: use constant from the previous comment

**Review by unreal998 — APPROVED** [review]

**Pull request merged**